set-definitions/src/main/java/eu/europeana/set/definitions/json/AgentDeserializer.java

```diff
@@ -0,0 +1,36 @@
+package eu.europeana.set.definitions.json;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import eu.europeana.set.definitions.model.agent.Agent;
+import eu.europeana.set.definitions.model.agent.impl.Person;
+import eu.europeana.set.definitions.model.agent.impl.SoftwareAgent;
+import eu.europeana.set.definitions.model.vocabulary.AgentTypes;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+
+public class AgentDeserializer extends JsonDeserializer<Agent> {
```

Comment:
this class should be moved to set-client, it is used only by the client

Reply:
done